File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/repository/ItemRepository.java
package pl.udu.uwr.pum.shoppyjava.repository;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

import pl.udu.uwr.pum.shoppyjava.data.Item;
import pl.udu.uwr.pum.shoppyjava.data.ItemDao;
import pl.udu.uwr.pum.shoppyjava.data.ItemDatabase;

public class ItemRepository {

    private final ItemDao itemDao;
    private final LiveData<List<Item>> readAllData;

    public ItemRepository(Application application){
        ItemDatabase db = ItemDatabase.getDatabase(application);
        itemDao = db.itemDao();
        readAllData = itemDao.readAllData();
    }

    public void insert(Item item) {
        ItemDatabase.databaseWriteExecutor.execute(() -> itemDao.addItem(item));
    }

    public LiveData<List<Item>> ReadAllData() {
        return readAllData;
    }
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/data/ItemRepository.kt
package pl.udu.uwr.pum.shoppykotlin.data

import androidx.lifecycle.LiveData

class ItemRepository(private val itemDao: ItemDao) {
    val readAllData: LiveData<List<Item>> = itemDao.readAllData()

    suspend fun addItem(item: Item){
        itemDao.addItem(item)
    }
}